SKR 1.4 -not compiling when choosing character based LCD

[Archiv.zip](https://github.com/MarlinFirmware/Marlin/files/4926236/Archiv.zip)

I am running an SKR 1.4 Turbo board in a Bear converted Prusa MK3 printer. I wanted to merge the latest bugfix 2.0.x changes into my local branches. When trying to compile, I get the following errors:

In file included from Marlin/src/HAL/LPC1768/../../core/../inc/../pins/lpc1769/pins_BTT_SKR_V1_4_TURBO.h:30,
                 from Marlin/src/HAL/LPC1768/../../core/../inc/../pins/pins.h:410,
                 from Marlin/src/HAL/LPC1768/../../core/../inc/MarlinConfig.h:32,
                 from Marlin/src/HAL/LPC1768/../../core/serial.h:24,
                 from Marlin/src/HAL/LPC1768/DebugMonitor.cpp:25:
Marlin/src/HAL/LPC1768/../../core/../inc/../pins/lpc1769/../lpc1768/pins_BTT_SKR_V1_4.h:317:6: error: #error "Character LCD not yet supported for SKR 1.4."
  317 |     #error "Character LCD not yet supported for SKR 1.4."
      |      ^~~~~
In file included from Marlin/src/HAL/LPC1768/../../inc/../pins/lpc1769/pins_BTT_SKR_V1_4_TURBO.h:30,
                 from Marlin/src/HAL/LPC1768/../../inc/../pins/pins.h:410,
                 from Marlin/src/HAL/LPC1768/../../inc/MarlinConfig.h:32,
                 from Marlin/src/HAL/LPC1768/HAL.cpp:24:
Marlin/src/HAL/LPC1768/../../inc/../pins/lpc1769/../lpc1768/pins_BTT_SKR_V1_4.h:317:6: error: #error "Character LCD not yet supported for SKR 1.4."
  317 |     #error "Character LCD not yet supported for SKR 1.4."
......

However, when I change the LCD controller to a graphical LCD, the compilation ends flawlessly. 
### My Configurations

Attached is an archive with my config files

### Steps to Reproduce

1. Run compilation with #define REPRAP_DISCOUNT_SMART_CONTROLLER enabled in configuation.h 
**Expected behavior:** Compilation finishes correctly without errors (as before)
**Actual behavior:** Compilation fails with above error message.

#### Additional Information
When disabling #define REPRAP_DISCOUNT_SMART_CONTROLLER and enabling #define REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER --> all works fine.

--- a/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_3.h
+++ b/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_3.h
@@ -261,6 +261,10 @@
     #define LCD_PINS_ENABLE         EXPA1_03_PIN
     #define LCD_PINS_D4             EXPA1_05_PIN
 
+  #elif HAS_ADC_BUTTONS
+
+    #error "ADC BUTTONS do not work unmodifed on SKR 1.3, The ADC ports cannot take more than 3.3v."
+
   #else                                           // !CR10_STOCKDISPLAY
 
     #define LCD_PINS_RS             EXPA1_07_PIN

--- a/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
+++ b/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
@@ -316,6 +316,10 @@
 
     #error "ADC BUTTONS do not work unmodifed on SKR 1.4, The ADC ports cannot take more than 3.3v."
 
+  #elif HAS_CHARACTER_LCD
+
+    #error "Character LCD not yet supported for SKR 1.4."
+
   #endif
 
 #endif // HAS_SPI_LCD

--- a/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
+++ b/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
@@ -312,11 +312,7 @@
 
     #endif // !FYSETC_MINI_12864
 
-  #elif HAS_CHARACTER_LCD
-
-    #error "Character LCD not yet supported for SKR 1.4."
-
-  #endif
+  #endif // HAS_GRAPHICAL_LCD
 
 #endif // HAS_SPI_LCD
 

--- a/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
+++ b/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
@@ -312,10 +312,6 @@
 
     #endif // !FYSETC_MINI_12864
 
-  #elif HAS_ADC_BUTTONS
-
-    #error "ADC BUTTONS do not work unmodifed on SKR 1.4, The ADC ports cannot take more than 3.3v."
-
   #elif HAS_CHARACTER_LCD
 
     #error "Character LCD not yet supported for SKR 1.4."
@@ -323,6 +319,10 @@
   #endif
 
 #endif // HAS_SPI_LCD
+
+#if HAS_ADC_BUTTONS
+  #error "ADC BUTTONS do not work unmodifed on SKR 1.4, The ADC ports cannot take more than 3.3v."
+#endif
 
 //
 // Neopixel LED

--- a/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
+++ b/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
@@ -312,9 +312,9 @@
 
     #endif // !FYSETC_MINI_12864
 
-  #elif HAS_CHARACTER_LCD
+  #elif ENABLED(ADC_KEYPAD)
 
-    #error "Character LCD not yet supported for SKR 1.4."
+    #error "ADC KEYPADs do not work unmodifed on SKR 1.4, The ADC ports cannot take more than 3.3v."
 
   #endif
 

--- a/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
+++ b/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
@@ -312,9 +312,9 @@
 
     #endif // !FYSETC_MINI_12864
 
-  #elif ENABLED(ADC_KEYPAD)
+  #elif HAS_ADC_BUTTONS
 
-    #error "ADC KEYPADs do not work unmodifed on SKR 1.4, The ADC ports cannot take more than 3.3v."
+    #error "ADC BUTTONS do not work unmodifed on SKR 1.4, The ADC ports cannot take more than 3.3v."
 
   #endif
 

